File: kotlin/src/main/kotlin/com/joshy/hywrapper/model/skyblock/AuctionsEndedResponse.kt
package com.joshy.hywrapper.model.skyblock

import com.joshy.hywrapper.model.HypixelResponse
import com.joshy.hywrapper.model.RateLimit
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient

/**
 * Information about the recently ended auctions.
 *
 * @property success Whether the request was successful.
 * @property cause The cause of the error, if the request failed.
 * @property auctions A list of [EndedAuction] for the past minute.
 * */
@Serializable
data class AuctionsEndedResponse(
    override val success: Boolean? = null,
    override val cause: String? = null,
    val auctions: List<EndedAuction>? = null,
) : HypixelResponse {
    @Transient
    override var rateLimit: RateLimit? = null
}

/**
 * Information about an ended auction.
 *
 * @property auctionId UUID of the auction.
 * @property seller UUID of the seller.
 * @property sellerProfile UUID of the seller's profile.
 * @property buyer UUID of the buyer.
 * @property buyerProfile UUID of the buyer's profile.
 * @property timestamp Timestamp of when the auction ended or was purchased.
 * @property price The price the auction was or ended at.
 * @property bin Whether or not the auction was a BIN auction.
 * @property item Item data of the auctioned item.
 * */
@Serializable
data class EndedAuction(
    @SerialName("auction_id")
    val auctionId: String? = null,
    val seller: String? = null,
    @SerialName("seller_profile")
    val sellerProfile: String? = null,
    val buyer: String? = null,
    @SerialName("buyer_profile")
    val buyerProfile: String? = null,
    val timestamp: Long? = null,
    val price: Int? = null,
    val bin: Boolean? = null,
    @SerialName("item_bytes")
    val item: String? = null,
)

File: typescript/src/models/skyblock/AuctionsEndedResponse.ts
import { HypixelResponse } from "../../types";

/**
 * Information about the recently ended auctions.
 */
export interface AuctionsEndedResponse extends HypixelResponse {
    /** A list of ended auctions for the past minute. */
    auctions?: EndedAuction[];
}

/**
 * Information about an ended auction.
 */
export interface EndedAuction {
    /** UUID of the auction. */
    auction_id?: string;
    /** UUID of the seller. */
    seller?: string;
    /** UUID of the seller's profile. */
    seller_profile?: string;
    /** UUID of the buyer. */
    buyer?: string;
    /** UUID of the buyer's profile. */
    buyer_profile?: string;
    /** Timestamp of when the auction ended or was purchased. */
    timestamp?: number;
    /** The price the auction was or ended at. */
    price?: number;
    /** Whether or not the auction was a BIN auction. */
    bin?: boolean;
    /** Item data of the auctioned item. */
    item_bytes?: string;
}
